tests/load/filesystem/test_filesystem_common.py

```diff
@@ -176,3 +180,25 @@ def test_s3_wrong_client_certificate(default_buckets_env: str, self_signed_cert:
 
     with pytest.raises(SSLError, match="SSL: CERTIFICATE_VERIFY_FAILED"):
         print(filesystem.ls("", detail=False))
+
+
+def test_filesystem_destination_config_reports_unused_placeholders() -> None:
```

Comment:
you should test pipeline factory: if placeholders passed explicitly are merged with ones that are configured.

also no end to end test where the pipeline loads data and part of the layout are callbacks

Reply:
I am adding a e2e test for this rn